onnxruntime/contrib_ops/cuda/diffusion/group_norm_impl.cu

```diff
@@ -439,37 +626,53 @@
   params.c = num_channels;
   params.groups = num_groups;
   params.hw = params.h * params.w;
+
+  // This will allocate as many blocks as possible to partition HW.
```

Comment:
why as many blocks as possible? If too many blocks, will atomicAdd be a bottleneck?

Reply:
if hw is much larger than 1024, like 320*320, it should be fine.

Reply:
Added some comments